Code/Editor/Plugins/ComponentEntityEditorPlugin/UI/Outliner/OutlinerWidget.cpp

```diff
@@ -1335,14 +1335,22 @@ void OutlinerWidget::SetEditorUiEnabled(bool enable)
     EnableUi(enable);
 }
 
-void OutlinerWidget::EnteredComponentMode([[maybe_unused]] const AZStd::vector<AZ::Uuid>& componentModeTypes)
+void OutlinerWidget::OnEditorModeActivated(
+    [[maybe_unused]] const AzToolsFramework::ViewportEditorModesInterface& editorModeState, AzToolsFramework::ViewportEditorMode mode)
 {
-    EnableUi(false);
+    if (mode == AzToolsFramework::ViewportEditorMode::Component)
```

Comment:
nit: `switch` conversion again, I won't both mentioning it again, let you decide what you think works best

Reply:
as above 👍